core/src/main/java/ai/timefold/solver/core/impl/move/streams/generic/ListChangeMove.java

```diff
@@ -0,0 +1,178 @@
+package ai.timefold.solver.core.impl.move.streams.generic;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable;
+import ai.timefold.solver.core.preview.api.domain.metamodel.PlanningListVariableMetaModel;
+import ai.timefold.solver.core.preview.api.domain.metamodel.VariableMetaModel;
+import ai.timefold.solver.core.preview.api.move.MutableSolutionView;
+import ai.timefold.solver.core.preview.api.move.Rebaser;
+
+import org.jspecify.annotations.NonNull;
+
+/**
```

Comment:
The class is very well-documented. I missed these types of comments in the other move classes and data stream classes.